### ISSUE TITLE
Linear features inappropriately rendered in amenity-line layer

### ISSUE THREAD

**Issue by jeisenbe** [gravitystorm/openstreetmap-carto#4104]
### Expected behavior
- All linear features should be rendered prior to text labels, so that a text label is not cut in half by a line, and so the linear features are in proper layering order

### Actual behavior
- The `amenity-line` layer, which was originally for point icons only, is being used to render `leisure=track` and `attraction=water_slide` lines since PR #3443 and PR #3346

### Solution
- Move `attraction=water_slide` line rendering to `waterway-bridges` layer
- Move `leisure=track` line rendering to `roads_sql` layer, or for simplicity it could be added to aerialways - above roads fill but below bridges.
- Render the text labels in `text-line`

**Comment by dch0ph** [comment]
@imagico This was resolved by #5046.

**Comment by imagico** [comment]
Yes, it was.  The remaining features in `amenity-line` are point symbols, which would probably be good to combine with points and polygons (#5237).
